src/core/Core_result.ml

```diff
@@ -49,7 +49,20 @@ type 'a match_result = {
 [@@deriving show]
 
 type t = {
-  matches : Pattern_match.t list;
+  (* Why is this here?
```

Comment:
Maybe put
(* old: matches: Pattern_match.t list;
 * Why do we added the fixes here?
 *  ....
 *)

Reply:
BTW great comment.

Reply:
I would maybe add a few alt: comment, explaining design alternative here,
like the stuff you wrote in the PR description.